src/lib/provable/test/blake2b.unit-test.ts

```diff
@@ -0,0 +1,84 @@
+import { ZkProgram } from '../../proof-system/zkprogram.js';
+import { Bytes } from '../wrapped-classes.js';
+import { Gadgets } from '../gadgets/gadgets.js';
+import { blake2b as nobleBlake2b } from '@noble/hashes/blake2b';
+import { bytes } from './test-utils.js';
+import {
+  equivalentAsync,
+  equivalentProvable,
+} from '../../testing/equivalent.js';
+import { Random, sample } from '../../testing/random.js';
+import { expect } from 'expect';
+
+
+
+for (let { digest_length ,preimage, hash } of testVectors()) {
+  let actual = Gadgets.BLAKE2B.hash(Bytes.fromString(preimage), digest_length);
+  expect(actual.toHex()).toEqual(hash);
+}
+
+
+function testVectors() {
```

Comment:
where are some of these test vectors from? we should use some official ones also (if they arent the official ones!)

Reply:
[Official test vectors](https://github.com/BLAKE2/BLAKE2/blob/master/testvectors/blake2b-kat.txt) have  key parameter which I didn't implement. Therefore, I had to create test vectors manually.

Reply:
I would in any case include tests with longer preimages, to make sure the code works when two blocks are required (meaning t1 is nonzero). I believe the longer preimage in these tests was 43 bytes. Let's try with 128 for example.

Reply:
> Therefore, I had to create test vectors manually.

How?

Reply:
I hashed messages from other hash function unit tests using `nobleBlake2b`. Also, I added longer messages with @querolita 's recommendation.